.github/workflows/ci.yaml

```diff
@@ -0,0 +1,240 @@
+name: GitHub Actions CI
+
+# Cache is used for all Windows and macOS dependencies (size approx. 1230 * 2 + 1850 = 4310 MiB)
+# Cache is not used for Ubuntu builds, because it already has all dependencies except
+# the appropriate libtorrent version, which only takes 3-5 minutes to build from source anyway
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    types: [edited, opened, reopened, synchronize]
+    branches: [ master ]
+
+env:
+  # Qt: 5.15.0
+  # libtorrent: RC_1_2 HEAD, 1.2.10
+  VCPKG_COMMIT: 32eccc18191fbb57b159784a1724d2d00613ae82
+  VCPKG_DEST_MACOS: /Users/runner/qbt_tools/vcpkg
+  VCPKG_DEST_WIN: C:\qbt_tools\vcpkg
+  UBUNTU_LIBTORRENT_VERSION: libtorrent-1.2.10
+
+# NOTE: for client code built with >= C++14, libtorrent 1.2.x must be built with >= C++14 as well.
+# Compilers that don't default to >= C++14 must be overriden.
+# See https://github.com/arvidn/libtorrent/issues/4965.
+
+jobs:
+
+  ci_ubuntu:
+    name: Ubuntu
+
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, ubuntu-18.04]
+        qbt_gui: ["GUI=ON", "GUI=OFF"]
+      fail-fast: false
+
+    runs-on: ${{ matrix.os }}
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+    - name: checkout repository
+      uses: actions/checkout@v2.3.2
+
+    - name: install all build dependencies except libtorrent from Ubuntu repos
+      run: |
+        sudo apt install \
+          build-essential cmake git ninja-build pkg-config \
+          libssl-dev libgeoip-dev zlib1g-dev \
+          libboost-dev libboost-chrono-dev libboost-random-dev libboost-system-dev \
+          libqt5svg5-dev qtbase5-dev qttools5-dev
+
+    - name: install libtorrent from source
+      run: |
+        git clone https://github.com/arvidn/libtorrent && cd libtorrent
+        git checkout ${{ env.UBUNTU_LIBTORRENT_VERSION }}
+        cmake -B cmake-build-dir -G "Ninja" \
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+          -Ddeprecated-functions=OFF \
+          --graphviz=cmake-build-dir/target_graph.dot
+        cmake --build cmake-build-dir
+        sudo cmake --install cmake-build-dir --prefix /usr/local
+
+    - name: build qBittorrent
+      run: |
+        cmake -B build -G "Ninja" \
+          -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+          -D${{ matrix.qbt_gui }} \
+          -DVERBOSE_CONFIGURE=ON \
+          --graphviz=build/target_graph.dot
+        cmake --build build
+
+    - name: upload artifact as zip
+      uses: actions/upload-artifact@v2.1.3
+      with:
+        name: qBittorrent-CI-Ubuntu_${{ matrix.os }}-${{ matrix.qbt_gui }}
```

Comment:
@Chocobo1 

Well, this artifact is clearly labeled "Ubuntu" But anyway, the point of the artifacts produced by this workflow is not to be used as nightly builds for regular usage (for that, we can have a nightly builds workflows, like I have proposed in the past and am now developing an improved version of), but instead, for troubleshooting purposes.

For example, you can run `ldd` on the generated executable, check the generated dependencies graph, the generated `compile_commands.json` compilation database, etc.

Reply:
>For example, you can run ldd on the generated executable, check the generated dependencies graph, the generated compile_commands.json compilation database, etc.

I don't think you presented reasonable examples.
Why would one examine the binary generated for other distro than the user's one? and this is limited to ubuntu distro. I presume the user would be more interested in building the project himself in his distro and look at the result binary.
Why would one look at the dependencies graph from the CI? He should produce it on his own machine if he is really into it. 
Same goes for the compile_commands.json you mentioned.

>but instead, for troubleshooting purposes.

I still don't think those will aid any troubleshooting. And those artifacts still seems very rarely used, I would still prefer not uploading them.

Reply:
@Chocobo1 

> Why would one examine the binary generated for other distro than the user's one?

Maybe they happen to be working in a machine with a different distro, but are still interested in resolving a bug that's happening on Ubuntu? And what about users and contributors who actually use Ubuntu? This is certainly useful for them as a reference.

> and this is limited to ubuntu distro.

If you want more CI jobs for more distros, that can be added later.

> I presume the user would be more interested in building the project himself in his distro and look at the result binary.
Why would one look at the dependencies graph from the CI? He should produce it on his own machine if he is really into it.
Same goes for the compile_commands.json you mentioned.

CI is supposed to be the canonical reference; it's useful to compare to when troubleshooting. The `compile_commands.json` in particular is very useful to examine the actual command-lines used to compile the program, even for systems that you may not have access to in the moment or at all.

> I still don't think those will aid any troubleshooting. And those artifacts still seems very rarely used, I would still prefer not uploading them.

You're not forced to download them or look at them if you think you don't need them, but why remove them if they can be useful for other people? There is nothing to lose, storage is free (and if it stops being free, we'll change it then).

Reply:
>>Why would one examine the binary generated for other distro than the user's one?
>
>Maybe they happen to be working in a machine with a different distro, but are still interested in resolving a bug that's happening on Ubuntu? And what about users and contributors who actually use Ubuntu? This is certainly useful for them as a reference.

Eh... I won't be dreaming for unrealistic situations, lets be practical here, just remove it, now. If the need arises we can reintroduce it later.

>and this is limited to ubuntu distro.
>
>If you want more CI jobs for more distros, that can be added later.

No, that is not what I meant.

>CI is supposed to be the canonical reference;

Disagree, but lets leave it another time.

>it's useful to compare to when troubleshooting. The compile_commands.json in particular is very useful to examine the actual command-lines used to compile the program, even for systems that you may not have access to in the moment or at all.

How often is this file going to change between regular commits in qbt? Seldom. Then there is little point to store it for every commit. macOS is an exception since no one on the team have access to it.

Reply:
@Chocobo1 

> Eh... I won't be dreaming for unrealistic situations, lets be practical here, just remove it, now. If the need arises we can reintroduce it later.

> How often is this file going to change between regular commits in qbt? Seldom. Then there is little point to store it for every commit. macOS is an exception since no one on the team have access to it.

You are being too narrow minded. Just because _you_ don't see the benefits (or don't think _you_ will ever make use of them), does not mean they are not there. It does not cost you or anyone else anything extra to have this. Why do you keep insisting on removing it? You don't pay (both figuratively and in the literal sense) for what you don't use here.

>> CI is supposed to be the canonical reference;
>Disagree, but lets leave it another time.

So what exactly is CI supposed to be? It must be a guarantee that "things work" for certain standard setups/environments. What's the point of it if it has no bearing on reality? Someone who builds qBittorrent on Windows 10 should be able to look at the Windows CI run and figure out "hey, the CI is completing successfully, if I can't build, there must be a problem on my end - I'll use the CI as a reference/point of comparison to solve my issue".

Reply:
>Just because you don't see the benefits (or don't think you will ever make use of them), does not mean they are not there. 

Then present a real world useful example, not imaginary ones.

>But when it does change, it is extremely useful and valuable. It saves a lot time in troubleshooting and may help catch issues earlier.

I already said it can be introduced later when it really helps. https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491643951

>Why do you keep insisting on removing it? You don't pay (both figuratively and in the literal sense) for what you don't use here.

Those arguments doesn't really help here. You're here presenting a PR and trying to win the team members approval. If we can't compromise/converge our opinions then that is that..

Reply:
@Chocobo1 

> Then present a real world useful example, not imaginary ones.

This is not imaginary. Imagine any contributor wants to quickly check the artifacts produced for another system at any time. This gives us some flexibility. Personally I have diagnosed Windows compilation issues during my testing thanks to this. If this is valid for macOS, why not for other systems? We shouldn't make assumptions about what systems contributors have available to them at any given time.

The burden of proof is on you to explain why exactly it is better to _not_ make these artifacts available. Again, I've explain how it is useful to others. Just because they are not useful to you, that's not a good reason to remove this.

>I already said it can be introduced later when it really helps. #13288 (comment)

I already explained it already really helps.

> Those arguments doesn't really help here. You're here presenting a PR and trying to win the team members approval. If we can't compromise/converge our opinions then that is that..

What? My point is simply that the presence of these artifacts won't bother you in any way. They are useful for those who make use of them, and don't get in the way of anyone else. If you just want to look at the build logs, you just look at the build logs.

You're withholding your approval for stupid reasons. Seriously, I get the impression from this review that you didn't really bother to understand what is going on and why certain things are done in certain ways. Have you even looked at the example run and its jobs? https://github.com/FranciscoPombal/qBittorrent/actions/runs/223928661. Worse, you are actively dismissing other's workflows and pushing the for the removal of features that I have already proven to be useful.

Reply:
>You're withholding your approval for stupid reasons. 

Oh, so you're trying to force/provoke me into approval!?

>Seriously, I get the impression from this review that you didn't really bother to understand what is going on and why certain things are done in certain ways. Have you even looked at the example run and its jobs? https://github.com/FranciscoPombal/qBittorrent/actions/runs/223928661. 

And doubt a reviewer is not using his brain!?

>Worse, you are actively dismissing other's workflows and pushing the for the removal of features that I have already proven to be useful.

Sigh, I cannot continue to work with you this way... no point of me staying here any longer.

Reply:
@glassez
See this https://github.com/qbittorrent/qBittorrent/pull/13414#issuecomment-695784163?
FranciscoPombal disrespected my review/opinion and created a simplified version of my opinions to downplay my opinion (and presumably make fun of me along the way).
Really I don't see why I should work with that guy anymore. Given the situation now, I think I reserve the right to ignore him and all of his work. If you still think this PR has any value, just go merge it instead.

@FranciscoPombal 
Sorry, the way you acted is over the line, your [last reply](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491688420) with your tone is barely OK but splattering your non-sense outside is not. Our collaboration ends here.

Reply:
@Chocobo1 

Get off your high horse. This whole review of yours is a tremendous disrespect to my work from the start. Review comments such as https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491103877 prove you didn't bother to understand what was going on or the context up to this moment https://github.com/qbittorrent/qBittorrent/pull/13070.

You never showed any initiative to solve the broken CI problem, which was an urgent matter to resolve, and now that I come here with everything already done, you post a review with stupid nonsensical nitpicks. Additionally, you did very little to help expedite the PRs that this one depended on. You just swoop in at the last minute, post comments that show you haven't been keeping up with discussions, and I have to repeat everything all over again.

Your attitude towards other contributors such as me and their opinions are incredibly dismissive. In the past few months I have made many important contributions for the project. I helped solve many important bug reports, some of which ended up directly contributing to uncovering important underlying issues in libtorrent. Me and a few others are helping keep the issue tracker afloat in the current situation that the project is facing, and you just shit all over that with comments like https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491684430.

"You presented your opinions, OK"? What the FUCK is that? Me and others bust their asses off helping out people and solving bug reports, and you want to keep an unsustainable support policy for the project that results in us accruing technical debt, and making the issue tracker situation even more difficult than it already is. At the same time, you show little preoccupation in breaking recent setups (Ubuntu 18.04) for no good reason, and I have to waste a lot of time explaining why that is a bad idea, or even just inconsistent with your other opinions about OS support.

Seriously, who are you to say what is or isn't important in this CI script? You won't care about it, we all know it's gonna be me or others who are going to maintain it and fix it anyway. Your not gonna read users logs and help solve bugs. Let others contribute to the project in the ways that you won't.

Reply:
@Chocobo1 

>FranciscoPombal disrespected my review/opinion and created a simplified version of my words to downplay my viewpoint (and presumably make fun of me along the way).

You are free to point out how any of that is a misrepresentation of what you said.

Reply:
>Get off your high horse. This whole review of yours is a tremendous disrespect to my work from the start...

Bravo, now you showed your true self.

>You never showed any initiative to solve the broken CI problem, which was an urgent matter to resolve, 

Well, unlike you my limited free time and motivation is a factor, but I guess you care less about that.

>Seriously, who are you to say what is or isn't important in this CI script? 

Well at least the currently missing maintainer asked me (and some others) to help doing the reviews and merge them if they are decent enough. I'm not obliged to review/comment on every PR/issue, that wasn't on the contract. @glassez please also take note of my viewpoint.

Reply:
>You are free to point out how any of that is a misrepresentation of what you said.

Your disrespectfulness.

Reply:
@Chocobo1 

> Well, unlike you my limited free time and motivation is a factor, but I guess you care less about that.

Now _you_ just showed your true self. Your free time is not more valuable than that of others.

> I'm not obliged to review/comment on every PR/issue, that wasn't on the contract. 

Straw man. No one said that, and that was never the implication.

---

Let's get this back on topic, for the sake of the project. The artifacts are useful for some contributors, even if they are not useful to you. The comment about the libtorrent ABI issue should be mentioned in compile guides, but it is also very relevant where it is in the CI script. It quickly makes it clear that the issue is universal, not just a workaround for the CI environments.

Reply:
>>Well, unlike you my limited free time and motivation is a factor, but I guess you care less about that.
>
>Now you just showed your true self. Your free time is not more valuable than that of others.

[I guessed right, you don't care.](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491695453)

>>I'm not obliged to review/comment on every PR/issue, that wasn't on the contract.
>
>Straw man. No one said that, and that was never the implication.

Straw man what? Being led astray by your own imagination?
You are free to believe what you like, but what I said is the fact, **there was no such rule on the contract/from the start** when I accepted this role. So keep your asses off me.

>Let's get this back on topic, for the sake of the project. 

For the sake of your precious time spent on PR I believe. You showed your rude attitude and now expect me to just get back and play along, WTF?

>The artifacts are useful for some contributors, even if they are not useful to you. 

My last [comment](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491684032) still stands.

>The comment about the libtorrent ABI issue should be mentioned in compile guides, but it is also very relevant where it is in the CI script.

I don't deny it and I already told you to [make it shorter](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491137998). And... you asked me where is the better place, I [replied](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491642953) and you ignored it, what, were you just asking for fun?

Seriously, what is the point of getting my approval as [it is worthless to you](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491694579)? I'll give you a hint that it is still possible to get this PR merged in the current form when the maintainer is present, that is, if you can persuade him, probably won't be hard.

Reply:
@Chocobo1 
> [I guessed right, you don't care.](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491695453)

???

> Straw man what? Being led astray by your own imagination?
> You are free to believe what you like, but what I said is the fact, **there was no such rule on the contract/from the start** when I accepted this role. So keep your asses off me.

You are arguing about a point I never made, which is "you're obliged to review/comment on every PR/issue".

> > Let's get this back on topic, for the sake of the project.
> 
> For the sake of your precious time spent on PR I believe. You showed your rude attitude and now expect me to just get back and play along, WTF?

Just leave your grievances aside for a moment and get back on topic. I'm making an effort to brush aside your disrespect and abusive nitpicking, surely you can put up with my rude attitude.

> 
> > The artifacts are useful for some contributors, even if they are not useful to you.
> 
> My last [comment](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491684032) still stands.

No, it doesn't. I have even stated that this is personally useful to _me_, if not for anyone else. Is that not enough? If it caused you serious trouble to have this in, I would consider an alternative, or even removing it. But it doesn't, so why do you want to force me to remove something that is useful to me (and possibly to others)?

Contrary to what you might think now, I _would_ go out of my way to change it if it caused you trouble, because that is a way I show you respect. But you are not presenting valid reasons and forcing me to remove something that I make use of. That is something that I cannot respect, and which I interpret as disrespect toward me.

> > The comment about the libtorrent ABI issue should be mentioned in compile guides, but it is also very relevant where it is in the CI script.
> 
> I don't deny it and I already told you to [make it shorter](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491137998). And... you asked me where is the better place, I [replied](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491642953) and you ignored it, what, were you just asking for fun?

Do you mind giving a concrete suggestion? Personally, I don't see a way to further compress the comment while preserving its intended usefulness, and I don't feel like trying to guess what you want. Just state it clearly, if I disagree, I'll state clearly my alternative, and so forth until we reach an agreement.

I don't disagree with adding it to somewhere in the wiki as well, I'll do it eventually.

> Seriously, what is the point of getting my approval as [my role is worthless to you](https://github.com/qbittorrent/qBittorrent/pull/13288#discussion_r491694579)? I'll give you a hint that it is still possible to get this PR merged in the current form when the maintainer is present, that is, if you can persuade him, probably won't be hard.

Your approval is not worthless. The point of at least seeing these last discussions through is to expedite this is to improve the project, despite our own grievances and differences of opinion.

Reply:
>You are arguing about a point I never made, which is "you're obliged to review/comment on every PR/issue".

I'm merely telling a fact which might be something you didn't realize before.

>Just leave your grievances aside for a moment and get back on topic. I'm making an effort to brush aside your disrespect and abusive nitpicking, surely you can put up with my rude attitude.

Alright, but I'm noting that each reviewer has their own focus area and lax area and no one except the maintainer is except from this process, and this is how the current review process goes.

>No, it doesn't. I have even stated that this is personally useful to me, if not for anyone else. Is that not enough? 

I see no harm (little inconvenience at most) in adding this functionality a bit later, from my POV why not hold it for a while and see how it goes? 
From my POV taking resources but not putting it into use is wasteful.

>But you are not presenting valid reasons and forcing me to remove something that I make use of. That is something that I cannot respect, and which I interpret as disrespect toward me.

I think I already explained myself clear enough.
Each person has their own mind and thoughts, it is not always possible to see reasoning from others mind, even trying very hard. I faced similar situation numerous time when submitting PRs, I believe all team members have. I've learned not to take it personal as it is merely subjective opinions on matters. 

>>I don't deny it and I already told you to make it shorter. And... you asked me where is the better place, I replied and you ignored it, what, were you just asking for fun?
>
>Do you mind giving a concrete suggestion? 

I think my previous suggestions are concrete enough, I'll add here that IMO it is taking too many lines that it is distracting when reading the CI script and it should be moved near to the affected actions as said before.

>Your approval is not worthless. The point of at least seeing these last discussions through is to expedite this is to improve the project, despite our own grievances and differences of opinion.

I hope you also agree that reviewer comments (especially from team members) are not worthless. 
FYI, when I propose a PR, I find it useless to keep insisting my own reasoning when sufficient information is presented and the reviewer still thinks otherwise, in that case I can only trust the reviewer is knowing what he is doing. An accepted yet compromised PR is still better than a never-to-be-merged PR which contributes nothing to the project.
Also as you probably already known, the currently employed merge process requires 2 out of 3 approvals from selected team members, I believe this is deliberately designed to collect opinions from multiple trusted minds and have tighter quality control on the PR, this also means making compromises are not unexpected.

Reply:
@Chocobo1 

> I'm merely telling a fact which might be something you didn't realize before.

I understand it.
 
> Alright, but I'm noting that each reviewer has their own focus area and lax area and no one except the maintainer is except from this process, and this is how the current review process goes.

I realize there are certain unspoken "partitions" (some clearer than others) in the code base. For example, RSS is glassez's domain, while front-end and other WebUI stuff is Piccirello's, etc... Is CI supposed to be your "turf"? This has occurred organically, as far as I can tell. Perhaps we should clarify these things internally. But that is a discussion for another time.

> > No, it doesn't. I have even stated that this is personally useful to me, if not for anyone else. Is that not enough?
> 
> I see no harm (little inconvenience at most) in adding this functionality a bit later, from my POV why not hold it for a while and see how it goes?
> From my POV taking resources but not putting it into use is wasteful.

Well, from my POV, why not add this now, if it doesn't inconvenience anyone and is already useful for some. We'll see how it goes, and if there are problems, we remove it.

There is nothing being "wasted", as a public project we have "unlimited" artifact storage (and it's not like the usage that will result from this workflow is abusive of that, it's not like 6 10 GiB artifacts are uploaded each time) that does not interfere with other things.

> I think I already explained myself clear enough.
> Each person has their own mind and thoughts, it is not always possible to see reasoning from others mind, even trying very hard. I faced similar situation numerous time when submitting PRs, I believe all team members have. I've learned not to take it personal as it is merely subjective opinions on matters.

Right, it is natural to have differences of opinion. The thing is, why should your subjective feeling that this could be added a bit later trump my objective arguments explaining that it is already useful and that there is no objective reason not to include it now?

> > Do you mind giving a concrete suggestion?
> 
> I think my previous suggestions are concrete enough, I'll add here that IMO it is taking too many lines that it is distracting when reading the CI script and it should be moved near to the affected actions as said before.

OK, I already explained why I think it makes more sense for this to be at the top, but I'm willing to compromise here. How about:

```
# NOTE: Avoids a libtorrent ABI issue. See https://github.com/arvidn/libtorrent/issues/4965
```

on top of the `Force AppleClang ...` step? 1 line, doesn't extend past column 100 even when accounting for indentation.

> I hope you also agree that reviewer comments (especially from team members) are not worthless.

Of course.

> FYI, when I propose a PR, I find it useless to keep insisting my own reasoning when sufficient information is presented and the reviewer still thinks otherwise, in that case I can only trust the reviewer is knowing what he is doing. An accepted yet compromised PR is still better than a never-to-be-merged PR which contributes nothing to the project.

I want to get this merged ASAP.

The thing is - this comes back to the point I made above about my good, objective reason to include the artifact stuff now, rather than later, and the fact that there is no good reason not to do it. I already compromised in the matter of the comment, as there is some merit to your request - the usefulness of the comment diminishes a bit, but isn't completely lost by moving it and shortening it. However, in what concerns this discussion about artifacts, you must either accept that _I_ know what I'm doing, or present a good, objective reason why this shouldn't be included now.

Sorry, but I can't accept that the reason I remove it is purely "because I had to appease Chocobo1". I don't think this sets a good precedent, and I don't think it is a healthy way of handling collaboration in a project.

> Also as you probably already know, the current employed merge process requires 2 out of 3 approvals from selected team members, I believe this is deliberately designed to collect opinions from multiple trusted minds and have tighter quality control on the PR, this also means making compromises are not unexpected and in fact, happens very often.

The design of the system is good, I agree. But this triumvirate has been broken as of late - as we all know, the merging process has been a 2-man show for the last several months. Participation and input from 3 members with merging rights is essential; IMO, it can avoid situations like these. I do realize that there is nothing that can be done about this in the near future at least.

Reply:
>How about:# NOTE: Avoids a libtorrent ABI issue. See https://github.com/arvidn/libtorrent/issues/4965
>
>on top of the Force AppleClang ... step? 1 line, doesn't extend past column 100 even when accounting for indentation.

OK.

I reevaluated and now I think the ubuntu logs maybe are a bit useful, we should keep them, sorry for the inconveniences.

My viewpoint on the ubuntu binaries are still the same, we should still keep them out in this PR. But I'm having another idea that probably won't be easy but it would justify uploading them and you probably would like to investigate it in a follow up PR (so this one gets merged first). This is about linking statically with all of our direct dependencies (zlib, openssl, qt), in this way the binary is much more portable (still not entirely) across distros.

Reply:
@Chocobo1 

> My viewpoint on the ubuntu binaries are still the same, we should still keep them out in this PR.
But here is another idea/compromise to improve the situation here, it probably won't be easy but it would justify uploading them from my POV and you probably would like to investigate it in a follow up PR (so this one gets merged quickly). This is about linking statically with all of our direct dependencies (zlib, openssl, qt, etc), in this way the binary is much more (still not entirely) portable across distros and the upside is the audiences would be larger than just the ubuntu community.

Something like this? https://github.com/qbittorrent/qBittorrent/wiki/Compiliation:-Fully-static-binaries-on-Linux-(glibc-or-musl)
I agree that it would be a nice addition, definitely worth investigating later.

> As said before, you shouldn't make it personal, either on yourself or on others and this is really how the review process goes, by collecting everyone opinion (even the ones you take it personally) and both sides making compromises so the opinions may converge and the result quality is consistent/improved in the end.

:+1: 

I also changed the comment now, in the latest force push.

Reply:
>linking statically with all of our direct dependencies (zlib, openssl, qt, etc), in this way the binary is much more (still not entirely) portable across distros and the upside is the audiences would be larger than just the ubuntu community.

I don't think it's a good idea to build something on CI in a way that is unlikely to be used in real life. AFAIK, it is usually linked dynamically on Linux.
Build artifacts are not the purpose for which CI works. This is just a by-product that can be useful to someone.
Personally, I don't mind having it here if it doesn't cost us anything. But don't want too much from it.

Reply:
>I don't think it's a good idea to build something on CI in a way that is unlikely to be used in real life. AFAIK, it is usually linked dynamically on Linux.

Right, but the point is the current artifacts are only targeted to ubuntu users (and hardly useful IMO). If we want something really useful and universal then the only way is static builds (uncommon but viable).

Reply:
>current artifacts are only targeted to ubuntu users

It's just a side effect of the fact that we use Ubuntu here...
What I'm really worried about is whether we'll miss build error by having a static linking here? Well, that is, under some conditions, the static one is successful, but the dynamic one fails?

Reply:
>What I'm really worried about is whether we'll miss build error by having a static linking here? Well, that is, under some conditions, the static one is successful, but the dynamic one fails?

It would be very unlikely under normal conditions although it still depends on the build script.